compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETargetUtils.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "AIETargetUtils.h"
+
+#include "llvm/ADT/DenseMap.h"
+#include "mlir/IR/BuiltinTypes.h"
+
+namespace mlir::iree_compiler::AMDAIE {
+
+namespace {
+
+// Generate a DenseMap key we can use for the element types (alternatives
+// considered: implement tombstone for std::array, or use std::map instead of
+// DenseMap).
+auto getElementTypeKey = [](uint32_t a, uint32_t b, uint32_t c) -> uint32_t {
+  return a + (b << 8) + (c << 16);
+};
+
+// Map from (LHS bitwidth, RHS bitwidth, Accumulator bitwidth) to the AIE
+// instruction size (m, n, k) for the integer types with those bitwidths.
+const auto& getIntegerMatmulInstructionSizeMap() {
+  // Sanity check.
+  static_assert(getElementTypeKey(1, 2, 3) == 1 + 2 * 256 + 3 * 65536);
+
+  static DenseMap<uint32_t, std::array<uint32_t, 3>> matmulIntSizes{
```

Comment:
Nice! but I think the key can be an `std::array` as well and it effectively hashes the elements (I think). If it diesnt work, this is fine as well.

Reply:
It can't, because the keys of DenseMaps need to define a tombstone key...